src/main/java/org/prebid/server/validation/RequestValidator.java

```diff
@@ -320,26 +320,19 @@ private void validateExtBidPrebidData(ExtRequestPrebidData data, Map<String, Str
 
     private void validateEidPermissions(List<ExtRequestPrebidDataEidPermissions> eidPermissions,
                                         Map<String, String> aliases) throws ValidationException {
-        if (eidPermissions != null) {
-            final Set<String> uniqueEidsSources = new HashSet<>();
-            for (ExtRequestPrebidDataEidPermissions eidPermission : eidPermissions) {
-                validateEidPermission(eidPermission, aliases, uniqueEidsSources);
-            }
-        }
-    }
 
-    private void validateEidPermission(ExtRequestPrebidDataEidPermissions eidPermission,
-                                       Map<String, String> aliases,
-                                       Set<String> uniqueEidsSources)
-            throws ValidationException {
-        if (eidPermission == null) {
-            throw new ValidationException("request.ext.prebid.data.eidpermissions[] can't be null");
+        if (eidPermissions == null) {
+            return;
         }
-        final String eidPermissionSource = eidPermission.getSource();
 
-        validateEidPermissionSource(eidPermissionSource);
-        validateDuplicatedSources(uniqueEidsSources, eidPermissionSource);
-        validateEidPermissionBidders(eidPermission.getBidders(), aliases);
+        for (ExtRequestPrebidDataEidPermissions eidPermission : eidPermissions) {
+            if (eidPermission == null) {
+                throw new ValidationException("request.ext.prebid.data.eidpermissions[] can't be null");
```

Comment:
Is it actually technically possible that here eidPermission will be null?

Also please update message with [I], because now it looks like array should not be null

Reply:
there is a unit test that covers that case, so technically yes, but from the logic perspective I'm not sure